fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/BiomeNeighborsPredicate.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.biomes.v1;
+
+import java.util.Collection;
+
+import com.google.common.collect.Lists;
+
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.layer.util.LayerRandomnessSource;
+
+/**
+ * A predicate of a biome collection and randomness source. Tests whether the conditions for generating the biome are met.
+ */
+@FunctionalInterface
+public interface BiomeNeighborsPredicate {
+	/**
+	 * @param biomes a collection of biomes that neighbor the biome to be replaced. May contain duplicates, depending on the biome api method with which this is used.
+	 * @param rand the world gen randomness source
+	 * @return whether the conditions for generation are met
+	 */
+	boolean test(Collection<Biome> biomes, LayerRandomnessSource rand);
```

Comment:
I think the class name and this method name should more closely reflect what it does: determine whether the biome this was registered for needs to be replaced. Predicate/test are awfully generic for such a specific use.

Additionally this is always an operation on the 4 adjacent biomes, so passing those 4 biomes directly seems preferable, similar to the signature of EaseBiomeEdgeLayer.sample. Using Biome over the biome id is a little slower, but avoids bugs with stale caches from biome id remapping, so it's probably a good thing.